test/hisi_zip_test/test_lib.c

```diff
@@ -41,6 +42,239 @@ void *mmap_alloc(size_t len)
 	return p == MAP_FAILED ? NULL : p;
 }
 
+static void gen_random_data(void *buf, size_t len)
+{
+	int i;
+	uint32_t seed = 0;
+	unsigned short rand_state[3] = {(seed >> 16) & 0xffff, seed & 0xffff,
+					0x330e};
+
+	for (i = 0; i < len >> 3; i++)
+		*((uint64_t *)buf + i) = nrand48(rand_state);
+}
+
+static int calculate_md5(comp_md5_t *md5, const void *buf, size_t len)
+{
+	if (!md5 || !buf || !len)
+		return -EINVAL;
+	MD5_Init(&md5->md5_ctx);
+	MD5_Update(&md5->md5_ctx, buf, len);
+	MD5_Final(md5->md, &md5->md5_ctx);
+	return 0;
+}
+
+static void dump_md5(comp_md5_t *md5)
+{
+	int i;
+
+	for (i = 0; i < MD5_DIGEST_LENGTH - 1; i++)
+		printf("%02x-", md5->md[i]);
+	printf("%02x\n", md5->md[i]);
+}
+
+static int cmp_md5(comp_md5_t *orig, comp_md5_t *final)
+{
+	int i;
+
+	if (!orig || !final)
+		return -EINVAL;
+	for (i = 0; i < MD5_DIGEST_LENGTH; i++) {
+		if (orig->md[i] != final->md[i]) {
+			printf("Original MD5: ");
+			dump_md5(orig);
+			printf("Final MD5: ");
+			dump_md5(final);
+			return -EINVAL;
+		}
+	}
+	return 0;
+}
+
+static int chunk_deflate(void *in, void *out, struct test_options *opts)
+{
+	size_t chunk_sz = opts->block_size;
+	int alg_type = opts->alg_type;
+	z_stream strm;
+	int windowBits;
+	int ret;
+
+	switch (alg_type) {
+	case WD_ZLIB:
+		windowBits = 15;
+		break;
+	case WD_DEFLATE:
+		windowBits = -15;
+		break;
+	case WD_GZIP:
+		windowBits = 15 + 16;
+		break;
+	default:
+		printf("algorithm %d unsupported by zlib\n", alg_type);
+		return -EINVAL;
+	}
+	memset(&strm, 0, sizeof(z_stream));
+	strm.next_in = in;
+	strm.avail_in = chunk_sz;
+	strm.next_out = out;
+	strm.avail_out = chunk_sz * EXPANSION_RATIO;
+
+	ret = deflateInit2(&strm, Z_BEST_SPEED, Z_DEFLATED, windowBits,
+			   8, Z_DEFAULT_STRATEGY);
+	if (ret != Z_OK) {
+		printf("deflateInit2: %d\n", ret);
+		return -EINVAL;
+	}
+
+	do {
+		ret = deflate(&strm, Z_FINISH);
+		if ((ret == Z_STREAM_ERROR) || (ret == Z_BUF_ERROR)) {
+			printf("defalte error %d - %s\n", ret, strm.msg);
+			ret = -ENOSR;
+			break;
+		} else if (!strm.avail_in) {
+			if (ret != Z_STREAM_END)
+				printf("deflate unexpected return: %d\n", ret);
+			ret = 0;
+			break;
+		} else if (!strm.avail_out) {
+			printf("deflate out of memory\n");
+			ret = -ENOSPC;
+			break;
+		}
+	} while (ret == Z_OK);
+
+	deflateEnd(&strm);
+	return ret;
+}
+
+/*
+ * This function is used in BLOCK mode. Each compressing in BLOCK mode
+ * produces compression header.
+ */
+static int chunk_inflate(void *in, void *out, struct test_options *opts)
```

Comment:
could we reuse what Jean had already done? I mean the software deflate/inflate function?